File: 02_Object_Oriented_Programming/05_Polymorphism_and_It's_types/01_Compile_Time_Polymorphism/02_Example_2/python/vehicle.py
class Vehicle:
    # Python simulates method overloading by checking argument types
    def start(self, vehicle_type_or_id):
        if isinstance(vehicle_type_or_id, str):
            print(f"Starting a {vehicle_type_or_id}")
        elif isinstance(vehicle_type_or_id, int):
            print(f"Starting a vehicle with ID: {vehicle_type_or_id}")

File: 02_Object_Oriented_Programming/05_Polymorphism_and_It's_types/01_Compile_Time_Polymorphism/02_Example_2/javascript/vehicle.js
class Vehicle {
    // JavaScript simulates method overloading by checking argument types
    start(vehicleTypeOrId) {
        if (typeof vehicleTypeOrId === 'string') {
            console.log(`Starting a ${vehicleTypeOrId}`);
        } else if (typeof vehicleTypeOrId === 'number') {
            console.log(`Starting a vehicle with ID: ${vehicleTypeOrId}`);
        }
    }
}

module.exports = Vehicle;

